src/OpenColorIO/ops/fixedfunction/FixedFunctionOpCPU.cpp

```diff
@@ -653,6 +690,145 @@ void Renderer_ACES_DarkToDim10_Fwd::apply(const void * inImg, void * outImg, lon
     }
 }
 
+float compress(float dist, float thr, float scale, float power, float one_over_power)
+{
+    // Normalize distance outside threshold by scale factor
+    const float nd = (dist - thr) / scale;
+    const float p = powf(nd, power);
+
+    // Compress
+    return thr + scale * nd / (powf(1.0f + p, one_over_power));
+}
+
+float uncompress(float dist, float thr, float scale, float power, float one_over_power)
+{
+    // Avoid singularity
+    if (dist >= (thr + scale))
+    {
+        return dist;
+    }
+    else
+    {
+        // Normalize distance outside threshold by scale factor
+        const float nd = (dist - thr) / scale;
+        const float p = powf(nd, power);
+
+        return thr + scale * powf(-(p / (p - 1.0f)), one_over_power);
+    }
+}
+
+template <typename Func>
+float gamut_comp(float val, float ach, float lim, float thr, float scale, float power, float one_over_power, Func f)
+{
+    // Disable compression
+    if (ach == 0.0f)
+    {
+        return 0.0f;
+    }
+
+    // Disable compression, avoid nan
+    if (lim < 1.0001f)
+    {
+        return val;
+    }
+
+    // Distance from the achromatic axis, aka inverse RGB ratios
+    const float dist = (ach - val) / std::fabs(ach);
+
+    // No compression below threshold
+    if (dist < thr)
+    {
+        return val;
+    }
+
+    // Compress / Uncompress distance with parameterized shaper function
+    const float comprDist = f(dist, thr, scale, power, one_over_power);
+
+    // Recalculate RGB from compressed distance and achromatic
+    const float compr = ach - comprDist * std::fabs(ach);
+
+    return compr;
+}
+
+Renderer_ACES_GamutComp13_Fwd::Renderer_ACES_GamutComp13_Fwd(ConstFixedFunctionOpDataRcPtr & data)
+    :   OpCPU()
+{
+    m_limCyan        = (float)data->getParams()[0];
+    m_limMagenta     = (float)data->getParams()[1];
+    m_limYellow      = (float)data->getParams()[2];
+    m_thrCyan        = (float)data->getParams()[3];
+    m_thrMagenta     = (float)data->getParams()[4];
+    m_thrYellow      = (float)data->getParams()[5];
+    m_power          = (float)data->getParams()[6];
+
+    m_one_over_power = 1.f / m_power;
```

Comment:
Minor: This seems like a lot of work to avoid a single divide, given all the other math that is happening.  I know Patrick commented on it, but given that you already tested it didn't help performance to precompute it, perhaps it's cleaner to just do the divide?  (Again, feel free to ignore this comment if you prefer to leave it as is.)

Reply:
Yes agreed, it feels a bit like premature optimisation and it could very well be that most compilers are able to fold this into a constant expression. Will revert for the time being.